include/fdeep/recurrent_ops.hpp

```diff
@@ -0,0 +1,184 @@
+// Copyright 2016, Tobias Hermann.
+// https://github.com/Dobiasd/frugally-deep
+// Distributed under the MIT License.
+// (See accompanying LICENSE file or at
+//  https://opensource.org/licenses/MIT)
+
+#pragma once
+
+#include <string>
+#include <functional>
+
+namespace fdeep { namespace internal
+{
+
+static float_type linear_activation(float_type x)
+{
+    return x;
+}
+
+static float_type tanh_activation(float_type x)
+{
+    return std::tanh(x);
+}
+
+static float_type sigmoid_activation(float_type x)
```

Comment:
Are these ops the same ops as the non-recursive ops currently coded in the layers? If so, we might unify them to avoid code duplication.

Reply:
But maybe we can do this in a separate clean-up PR after the new functionality here is merged.

Reply:
Those are exactly the same, so unifying them would definitely make sense.